polkadot-host-spec/c03-transition.tm

```diff
@@ -670,112 +671,198 @@
+
+    \;
+  <|big-table>
+    <label|table-changes-trie-key-types>Possible types of keys of mappings in
+    the Changes Trie
+  </big-table>
+
+  <strong|Note>: Unlike the default encoding for varying data types, this
+  structure starts its indexing at <verbatim|1>.
+
+  <\definition>
+    <label|defn-storage-key-to-extrinsics>The <strong|storage key to
+    extrinsics mappings> track any changes which occure in an individual
+    block. Its value is a SCALE encoded array containing the indices of the
+    extrinsics that caused any changes to its key. The indices are unsigned
+    32-bit integers and their values is based on the order in which each
+    extrinsic is appears in the block. Additionally, a separate Changes Trie
+    is created for each child storage where changes occured.
+
+    \;
+
+    Keys that were modified outside of extrinsics, such as changes made by
+    the Runtime directly, have a <verbatim|0xffffffff> value as its indice.
+
+    \;
+
+    The Changes Trie is generated when the Runtime calls
+    <verbatim|ext_storage_changes_root> as described in section
+    <reference|sect-ext-storage-changes-root>.
+  </definition>
+
+  <\definition>
+    <label|defn-storage-key-to-blocks>The <strong|storage key to blocks
+    mappings> track any changes which occured in a certain range of blocks.
+    Its value is a SCALE encoded array containing block numbers where
+    extrinsics caused any changes to its key. The block numbers are
+    represented as unsigned 32-bit integers.
+
+    \;
+
+    The Changes Trie is generated when the Runtime calls
+    <verbatim|ext_storage_changes_root> as described in section
+    <reference|sect-ext-storage-changes-root>, but unlike the other mappings,
+    those are not generated on each block.
+  </definition>
+
+  <\definition>
+    <label|defn-storage-key-to-child-tries>The <strong|storage key to Child
+    Changes Tries mappings> track any storage keys from child storages and
+    their corresponding Child Changes Trie. As described in Definition
+    <reference|defn-storage-key-to-extrinsics>, changes inside child storages
+    are inserted into their own Trie. The changed child storage key is
+    inserted into the main Trie, where its value is the Merkle proof as
+    desribed in section <reference|sect-merkl-proof> of the corresponding
+    Child Storage Trie.
+  </definition>
+
+  The Changes Trie itself is not part of the block, but a separately
+  maintained database by the Polkadot node. The Merkle proof of the Changes
+  Trie must be inlcuded in the block digest as described in Definition
+  <reference|defn-digest> and gets calculated as described in section
+  <reference|sect-merkl-proof>. The root calculation uses entries of a
+  per-block basis (including block mappings when conditions apply on that
+  block), not the whole changes spanning multiple blocks.
```

Comment:
The meaning of this is not quite clear. For a block where block mappings are generated the changes trie must include entries related to any extrinsic changed in the tracked range of blocks, right? What does per-block basis mean in this case?